### PR TITLE
Add benchmark script and results to documentation

### PR DESCRIPTION
**Pull request by IgorSusmelj**
### Changes

- Added benchmark script used to generate the table in the README
- Added a more detailed overview of the benchmarking procedure, results as well as plots

### Preview
The new documentation page looks like this:
[Benchmarks — lightly 1.0.0 documentation.pdf](https://github.com/lightly-ai/lightly/files/5826378/Benchmarks.lightly.1.0.0.documentation.pdf)

### CODE REVIEW

File: docs/source/getting_started/benchmarks.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+You can reproduce the benchmarks using the following script:
+:download:`cifar10_benchmark.py <benchmarks/cifar10_benchmark.py>` 
+
+.. figure:: images/bench_knn-cifar10-200epochs.png
+    :align: center
+    :alt: kNN accuracy on test set for training the models for 200 epochs
+
+    kNN accuracy on test set for training the models for 200 epochs. The shorter
+    runs (less steps) are the ones with batch size of 512.
+
```
Comment (philippmwirth) [inline docs/source/getting_started/benchmarks.rst:48]: I would say "fewer steps" rather than "less steps". I'm also missing a legend (you could simply add it to the description).

File: docs/source/getting_started/benchmarks.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Self-Supervised models benefit from long training times.
+
+.. figure:: images/bench_knn_accuracy_cifar10_800epochs.png
+    :align: center
+    :alt: kNN accuracy after 800 epochs 
+
+    kNN accuracy after 800 epochs. Red: SimSiam, Orange: MoCo, Blue: SimCLR 
+
+The loss behaves differently for the various models. Whereas MoCo and SimCLR use
+a contrastive loss SimSiam uses negative cosing similarity.
```
Comment (philippmwirth) [inline docs/source/getting_started/benchmarks.rst:68]: "Noticeably, the methods which use a contrastive loss (MoCo and SimCLR) behave similar, making quick accuracy improvements in the beginning but then slowing down. In contrast, SimSiam with the negative cosine similarity loss  takes longer to warm up but then catches up quickly to the others."

?

File: docs/source/getting_started/benchmarks.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+.. figure:: images/bench_train_loss_cifra10_800_epochs.png
+    :align: center
+    :alt: Training loss for 800 epochs
+
```
Comment (philippmwirth) [inline docs/source/getting_started/benchmarks.rst:73]: I'm not sure this plot is too interesting :/

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        shuffle=False,
+        drop_last=False,
+        num_workers=num_workers
+    )
+
+    return dataloader_train_ssl, dataloader_train_kNN, dataloader_test
+
+# code from here:
+# https://colab.research.google.com/github/facebookresearch/moco/blob/colab-notebook/colab/moco_cifar10_demo.ipynb
+
```
Comment (philippmwirth) [inline docs/source/getting_started/benchmarks/cifar10_benchmark.py:134]: which code are your referring to?

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_simsiam.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+
+model_names = ['SimCLR_512', 'SimSiam_512']
+models = [SimCLRModel, SimSiamModel]
```
Comment (philippmwirth) [inline docs/source/getting_started/benchmarks/cifar10_benchmark.py:332]: You're using batch sizes `128` and `512` yet the model names say `512`. To make the scripts safe, should we not set the model names dynamically? Also, where is MoCo? I would add it to the list of models.

**Review by philippmwirth — CHANGES_REQUESTED** [review]

File: docs/source/getting_started/benchmarks.rst (hunk 3 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/benchmarks.rst:73]: I left it there to show how the plot evolves during training. I like to have a reference point on how the plot could look like.

File: docs/source/getting_started/benchmarks.rst (hunk 1 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/benchmarks.rst:49]: done

File: docs/source/getting_started/benchmarks.rst (hunk 2 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/benchmarks.rst]: done

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py (hunk 5 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/benchmarks/cifar10_benchmark.py]: done

File: docs/source/getting_started/benchmarks/cifar10_benchmark.py (hunk 4 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/benchmarks/cifar10_benchmark.py]: I updated the comment

**Review by philippmwirth — APPROVED** [review]
